UPGRADE.md

```diff
@@ -0,0 +1,61 @@
+### Kubevirt HCO upgrade procedure
+
```

Comment:
Let's create a section to describe what happens when you set upgrade to automatic.  Then, the next few sections will be for manual upgrade.

Reply:
Pushed an updated version.

Reply:
I think we should reference [the OLM docs](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/how-to-update-operators.md) when talking about upgrade and avoid duplication where possible.